src/support/z_template.erl

```diff
@@ -104,31 +104,17 @@ render_block(OptBlock, Template, Vars, Context) when is_map(Vars) ->
     case Result of
         {ok, Output} ->
             Output;
-        {error, {{ErrFile,Line,Col}, _YeccModule, Error}} ->
-            try
-                Error1 = iolist_to_binary(Error),
-                lager:error("[~p] Error rendering template: ~s:~p (~s)~n",
-                           [z_context:site(Context), ErrFile, Line, Col, Error1])
-            catch
-                _:_ ->
-                    lager:error("[~p] Error rendering template: ~s:~p (~p)~n",
-                               [z_context:site(Context), ErrFile, Line, Col, Error])
-            end,
+        {error, {{ErrFile, Line, Col}, _YeccModule, Error}} ->
+            lager:error(
+                "Error rendering template ~s:~p:~p due to ~s ~n",
+                [ErrFile, Line, Col, Error]
+            ),
             <<>>;
         {error, Reason} when is_list(Reason); is_binary(Reason) ->
-            try
-                Reason1 = iolist_to_binary(Reason),
-                lager:error("[~p] Error rendering template: ~s (~s)~n",
-                           [z_context:site(Context), Template, Reason1])
-            catch
```

Comment:
It is needed because the Reason might not be an `iolist()`, in that case we show it as a `~p` in the catch